### PR TITLE
Wishlist management

### PR DESCRIPTION
**Pull request by purvasharmadev**
## Changes made:

> added the wishlist feature i.e "add to wishlist", "remove from wishlist".

## Files to Check

> routes.js
> wishlist.jsx
> wishlist-context.js
> ProductCard.jsx

### CODE REVIEW

**Review by humzahasan — APPROVED** [review]
- Good Consistency in code
- Good use of custom hooks and abstracting of individual functions
- Appealing UI 

Good to merge!

File: src/Pages/Category.jsx [hunk 1]
```diff
@@ -1,20 +1,29 @@
-import React from 'react'
+import React from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useProducts } from "../Context/context";
+import { useParams } from "react-router-dom";
 
+// to={`/product/${props.id}`}
```
Comment (varun13169) [inline src/Pages/Category.jsx:6]: remove comment

File: src/Pages/ProductCard.jsx [hunk 2]
```diff
@@ -1,18 +1,57 @@
 import React from "react";
 import { useCart } from "../Context/cart-context";
+import { useList } from "../Context/wishlist-context";
```
Comment (varun13169) [inline src/Pages/ProductCard.jsx:3]: nit: useList -> useWhishlist

self-explanatory name.

File: src/Context/wishlist-context.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    } catch (error) {
+      console.log("error is :", error);
+    }
+  }
+
+  useEffect(() => {
+    if (ListItems.length > 0) {
+      return ListItems;
+    }
+  }, [ListItems]);
```
Comment (itsmebipul00) [inline src/Context/wishlist-context.js:54]: Nice that you are handling all edge cases

**Review by itsmebipul00 — APPROVED** [review]
Good to merge. Nice clean code.

**Review by adityapandadev — APPROVED** [review]
Clean Code, Good 👍

**Pull request merged**